### PR TITLE
python310Packages.git-url-parse: init at 1.2.2

### PR DESCRIPTION
**Pull request by blaggacao**
###### Description of changes

Python git url parser, part of `frappe/frappe` deps.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by blaggacao** [comment]
Result of `nixpkgs-review pr 244734` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.git-url-parse</li>
    <li>python311Packages.git-url-parse</li>
  </ul>
</details>

**Comment by teutat3s** [comment]
Result of `nixpkgs-review pr 244734` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.git-url-parse</li>
    <li>python310Packages.git-url-parse.dist</li>
    <li>python311Packages.git-url-parse</li>
    <li>python311Packages.git-url-parse.dist</li>
  </ul>
</details>

**Review by teutat3s — APPROVED** [review]

File: pkgs/development/python-modules/git-url-parse/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  src = fetchFromGitHub {
+    owner = "coala";
+    repo = "git-url-parse";
+    rev = version;
+    hash = "sha256-+0V/C3wE02ppdDGn7iqdvmgsUwTR7THUakUilvkzoYg=";
+  };
+
+  # Manually set version because prb wants to get it from the git
+  # upstream repository (and we are installing from tarball instead)
+  PBR_VERSION = version;
```
Comment (natsukium) [inline pkgs/development/python-modules/git-url-parse/default.nix:22]:
```suggestion
  env.PBR_VERSION = version;
```

File: pkgs/development/python-modules/git-url-parse/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  propagatedBuildInputs = [
+    pbr
+  ];
+
+  pythonImportsCheck = [ "giturlparse" ];
+
+  nativeCheckInputs = [
+    pytest
```
Comment (natsukium) [inline pkgs/development/python-modules/git-url-parse/default.nix:31]:
```suggestion
    pytestCheckHook
```
Would you please use `pytestCheckHook`?

Comment (blaggacao) [reply pkgs/development/python-modules/git-url-parse/default.nix]: eh sorry, I should have prepared these a bit more sequentially properly incorporating the learnings along the way. :handshake:

Comment (natsukium) [reply pkgs/development/python-modules/git-url-parse/default.nix]: Don't worry, I look forward to seeing your great progress.🥳

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 244734` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.git-url-parse</li>
    <li>python310Packages.git-url-parse.dist</li>
    <li>python311Packages.git-url-parse</li>
    <li>python311Packages.git-url-parse.dist</li>
  </ul>
</details>

**Comment by natsukium** [comment]
We do not need to measure coverage, so we should add a patch here.
https://github.com/coala/git-url-parse/blob/d3eac95b21b2b166562e657fdfd974545653dfcc/pytest.ini#L2C4-L2C4

**Comment by blaggacao** [comment]
Result of `nixpkgs-review pr 244734` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.git-url-parse</li>
    <li>python311Packages.git-url-parse</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

**Pull request merged**